src/curryN.js

```diff
@@ -39,8 +39,13 @@ module.exports = _curry2(function curryN(length, fn) {
         return arity(Math.max(length - (args && args.length || 0), 0), function() {
             if (arguments.length === 0) { throw _noArgsException(); }
             var newArgs = _concat(args, arguments);
+            var value;
             if (newArgs.length >= length) {
-                return fn.apply(this, newArgs);
+                value = fn.apply(this, newArgs);
+                if (typeof value === 'function') {
+                    return curryN(value.length, value);
```

Comment:
I believe this will return a broken function if `value.length` is 0. Perhaps `curryN` should return the provided function if `length` is 0 or 1. Alternatively it could throw if `length` is 0 rather than returning a function that will throw if called.

```javascript
R.curryN(0, function() { return 42; })()
// => TypeError: Function called with no arguments
```